src/OrleansAzureUtils/AzureSilo.cs

```diff
@@ -227,11 +228,29 @@ public void Run()
 
             if (host.IsStarted)
                 host.WaitForOrleansSiloShutdown();
-            
             else
                 throw new ApplicationException("Silo failed to start correctly - aborting");
         }
 
+		/// <summary>
+		/// Makes this Orleans silo begin executing and become active.
+		/// Note: This method call will only return control back to the caller when the silo is shutdown or 
+		/// an external request for cancellation has been issued.
+		/// </summary>
+		/// <param name="cancellationToken">Cancellation token.</param>
+		public void Run(CancellationToken cancellationToken)
```

Comment:
Can we eliminate the code duplication, by having Run and Run(CancellationToken) both call Run_Impl(CancellationToken) and check CancellationToken  for null inside.